Addition of Code of Conduct

kylemcdonald opened the pull request:
This PR adds a CODE_OF_CONDUCT.md alongside the README.md and LICENSE.md. @ofZach mentioned this one on Twitter and I thought it was nice, so I thought I'd add it.

Probably more important than any code of conduct itself is the discussion that happens around it. I brought this up in October and some of the comments then were:

- Make the "Philosophy" section more prominent on the "About" page (it's been moved to the top).
- It'd be nice, but we already have a lot of text on the site. We should organize and consolidate the text in a smart way (@workergnome started working on this, but it's on hiatus).
- Changing the language a little bit. For example, "If you are not a part of the group, welcome!" got changed to "If you are not yet...". Changing "of-dev" to "of-contributors" could also be good.
- Making it more clear how people can get involved should go along with any code of conduct. There's a short "Contribute" section under the "Community" page that @caitlinmorris added, but as far as I can tell no one has acted on any of the suggestions listed there. I think it might be time to try a new approach where we just have a single link that says "if you're interested in contributing, send an email to this person" and then we work it out from there. More personal relationships are probably going to work better than more diverse descriptions of ways to contribute.

Anyway, I wanted to post this PR as a way to keep this discussion going :)

ofZach commented:
+1 to this !  pinging @phoenixperry who originally suggested this on the unofficial OF slack...

arturoc commented:
+1 but although i know this is taken verbatim from the covenant code of conduct there's a couple of things that i would change. in the first paragraph not sure if the enumeration of activities makes much sense, it seems to be highly biased towards people doing developers tasks, i would change it to something like `involved in any way with the openFrameworks community` or similar.

in the 3rd paragraph at the end says unprofessional conduct which i would change with something else but would be ok with that too.

btw, the berlin code of conduct for meetups and other communities is quite nice: http://berlincodeofconduct.org/ it's more oriented to presencial events but seems to me to be somehow more open or oriented to more diverse communities than this one which sounds perhaps oriented to developer communities of mostly corporate software? just to have more examples on which to base ours

kylemcdonald commented:
@arturoc great suggestions, thanks for the feedback. i replaced "unprofessional" with "disrespectful", added the "involved in any way", and integrated a lot of ideas from the berlin code of conduct.

i'm going to send it to the of-dev list now for more feedback.

bakercp commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,20 @@
+* Attempt collaboration before conflict.
+* Be mindful of other community members.
+
+Alert anyone in [the core](https://github.com/orgs/openframeworks/teams/core) if you notice a dangerous situation, someone in distress, or violations of this Code of Conduct, even if they seem inconsequential.
```
[the core](https://github.com/orgs/openframeworks/teams/core) This link points to 404.

kylemcdonald replied:
thanks, i just realized the team members are not listed unless you're an owner. i'll fix this.

bilderbuchi replied:
maybe this can be fixed/corrected with the [new org permissions](https://github.com/blog/2020-improved-organization-permissions)?

lmccart commented:
Where it says "Alert any one of the...", how should they alert them? Is there an email address or should they ping them in the thread? I think it's important for people to know how (and feel comfortable and welcome) to express when they feel something's off.

kylemcdonald commented:
@lmccart i was also uncertain about this. the original text suggested opening an issue, but i don't think it's always appropriate (or comfortable) to do that. so i thought maybe a conduct at openframeworks.cc email would be good, but then maybe it feels too impersonal? i think most people don't want to post email  address in public -- but i could just post mine. what do you think?

lmccart commented:
for p5.js it's hello@p5js.org. we felt making a conduct@p5js.org or other seemed overly serious, we wanted to encourage people to say something even if it was just a minor weirdness they felt.

kylemcdonald commented:
great. i'll use of@openframeworks.cc which already goes to the core, and if someone @ofzach @oftheo @arturoc prefers another address, we can switch it out.
